API.md

```diff
@@ -0,0 +1,1246 @@
+        -   [value][21]
+        -   [increment][22]
+        -   [event][23]
+    -   [enqueue][24]
+-   [Infra][25]
+    -   [CacheAgent][26]
+    -   [InstrumentationAgent][27]
+    -   [QueueAgent][28]
+-   [Hull.Middleware][29]
+-   [Types][30]
+    -   [THullAccountAttributes][31]
+    -   [THullAccountIdent][32]
+    -   [THullAccount][33]
+    -   [THullAttributeName][34]
+    -   [THullAttributeValue][35]
+    -   [THullAttributesChanges][36]
+    -   [THullConnector][37]
+    -   [THullEvent][38]
+    -   [THullObjectAttributes][39]
+    -   [THullObjectIdent][40]
+    -   [THullObject][41]
+    -   [THullReqContext][42]
+    -   [THullSegment][43]
+    -   [THullSegmentsChanges][44]
+    -   [THullUserAttributes][45]
+    -   [THullUserChanges][46]
+    -   [THullUserIdent][47]
+    -   [THullUserUpdateMessage][48]
+    -   [THullUser][49]
+-   [Utils][50]
+    -   [notifHandler][51]
+    -   [oAuthHandler][52]
+    -   [smartNotifierHandler][53]
+    -   [superagentErrorPlugin][54]
+    -   [superagentInstrumentationPlugin][55]
+    -   [superagentUrlTemplatePlugin][56]
+
+## HullConnector
+
+**Parameters**
+
+-   `HullClient` **HullClient** 
+-   `options` **[Object][57]**  (optional, default `{}`)
+    -   `options.hostSecret` **[string][58]?** secret to sign req.hull.token
+    -   `options.port` **([Number][59] \| [string][58])?** port on which expressjs application should be started
+    -   `options.clientConfig` **[Object][57]?** additional `HullClient` configuration (optional, default `{}`)
+    -   `options.instrumentation` **[Object][57]?** override default InstrumentationAgent
+    -   `options.cache` **[Object][57]?** override default CacheAgent
+    -   `options.queue` **[Object][57]?** override default QueueAgent
+    -   `options.connectorName` **[string][58]?** force connector name - if not provided will be taken from manifest.json
+    -   `options.skipSignatureValidation` **[boolean][60]?** skip signature validation on notifications (for testing only)
+    -   `options.timeout` **([number][59] \| [string][58])?** global HTTP server timeout
+    -   `options.segmentFilterSetting`  
+
+### setupApp
+
+This method applies all features of `Hull.Connector` to the provided application:
+
+-   serving `/manifest.json`, `/readme` and `/` endpoints
+-   serving static assets from `/dist` and `/assets` directiories
+-   rendering `/views/*.html` files with `ejs` renderer
+-   timeouting all requests after 25 seconds
+-   adding Newrelic and Sentry instrumentation
+-   initiating the wole [Context Object][14]
+-   handling the `hullToken` parameter in a default way
+
+**Parameters**
+
+-   `app` **express** expressjs application
+
+Returns **express** expressjs application
+
+### startApp
+
+This is a supplement method which calls `app.listen` internally and also terminates instrumentation of the application calls.
+
+**Parameters**
+
+-   `app` **express** expressjs application
+
+Returns **http.Server** 
+
+## Errors
```

Comment:
Is it possible to reorder content?
I'd do:
- HullConnector
- Hull.Middleware
- Context
- Infra
- Utils
- Helpers (btw, seems redundant to have Utils & Helpers - but that's probably a refactor)
- Types
- Errors

Reply:
good one, I will set that order. Thanks for input.